src/app/(auth)/signup/page.tsx

```diff
@@ -29,49 +31,83 @@ export default function SignUp() {
     if (error) {
       throw new Error(`An error occurred trying to sign up: ${error.message}`);
     }
-    push('/');
+    setEmailSentCount(1);
+    // push('/verify-email');
+  };
+  const handleResendEmail = async () => {
+    const { error } = await supabase.auth.resend({
+      type: 'signup',
+      email,
+    });
+    if (error) {
+      throw new Error(
+        `An error occurred trying to resend email: ${error.message}`,
+      );
+    }
+    setEmailSentCount(emailSentCount + 1);
   };
 
   return (
     <OuterDiv>
-      <FormDiv>
-        <H1>Sign Up</H1>
-        <SpacerDiv>
-          <QuestionsDiv>
-            <TextInput
-              label="Email"
-              placeholder="email@example.com"
-              erroring={false}
-              errorText="Email Error" // {errorMessage}
-              type="email"
-              name="email"
-              value={email}
-              setValue={setEmail}
-            />
-            <TextInput
-              label="Password"
-              placeholder="Password"
-              erroring={false}
-              errorText="Password Error"
-              type="password"
-              name="password"
-              value={password}
-              setValue={setPassword}
-            />
-          </QuestionsDiv>
-        </SpacerDiv>
-        <SpacerDiv>
-          <BigButton type="button" onClick={handleSignUp}>
-            <H4 $color="white">Sign Up</H4>
-          </BigButton>
-          <H4Centered>
-            Have an account already?{' '}
-            <AColored $color={COLORS.greyDark} href="/login">
-              Log In
-            </AColored>
-          </H4Centered>
-        </SpacerDiv>
-      </FormDiv>
+      {!emailSentCount && (
+        <FormDiv>
+          <H1>Sign Up</H1>
+          <TextInput
+            label="Email"
+            placeholder="email@example.com"
+            erroring={false}
+            errorText="Email Error" // {errorMessage}
+            type="email"
+            name="email"
+            value={email}
+            setValue={setEmail}
+          />
+          <TextInput
+            label="Password"
+            placeholder="Password"
+            erroring={false}
+            errorText="Password Error"
+            type="password"
+            name="password"
+            value={password}
+            setValue={setPassword}
+          />
+          <SpacerDiv>
+            <BigButton type="button" onClick={handleSignUp}>
```

Comment:
Can you explain why you chose to pass in the function itself to `onClick` here, but went with `() => handleResendEmail()` on the other one? I think we would want the syntax to stay consistent.

Reply:
ahh i must have overlooked that-- i've changed it to just pass in the function itself. thanks!